ext/scripts/tests/install_scripts/run_pip_tests.sh

```diff
@@ -9,61 +9,151 @@ SCRIPT_DIR="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
 assert "$TEST_OUTPUT" "Dry-Run: python3 -m pip install  --no-cache-dir 'humanfriendly==9.1' 'requests==2.27.1' 'git+http://github.com/exasol/bucketfs-utils-python.git@0.2.0#egg=exasol-bucketfs-utils-python'"
 echo
 
+
 echo ./install_via_pip.pl with versions, with allow-no-version-for-urls, file with urls and some missing versions
-"$PATH_TO_INSTALL_SCRIPTS/install_via_pip.pl" --file test_files/pip/with_versions/with_urls_some_missing_versions --with-versions --allow-no-version-for-urls --python-binary python3 "$DRY_RUN_OPTION" || echo PASSED
+run_install "$PATH_TO_INSTALL_SCRIPTS/install_via_pip.pl" --file test_files/pip/with_versions/with_urls_some_missing_versions --with-versions --allow-no-version-for-urls --python-binary python3 "$DRY_RUN_OPTION" || echo PASSED
 echo
 
+
 echo ./install_via_pip.pl with pip version syntax
-TEST_OUTPUT=$("$PATH_TO_INSTALL_SCRIPTS/install_via_pip.pl" --file test_files/pip/pip_version_syntax --python-binary python3 "$DRY_RUN_OPTION")
+TEST_OUTPUT=$(run_install "$PATH_TO_INSTALL_SCRIPTS/install_via_pip.pl" --file test_files/pip/pip_version_syntax --python-binary python3 "$DRY_RUN_OPTION")
 assert "$TEST_OUTPUT" "Dry-Run: python3 -m pip install  --no-cache-dir 'humanfriendly==9.1' 'requests>=2.21.0' 'git+http://github.com/exasol/bucketfs-utils-python.git@0.2.0#egg=exasol-bucketfs-utils-python'"
 echo
 
+
+echo ./install_via_pip.pl installing a package twice with different versions must fail
+TEST_OUTPUT=$(run_install_must_fail "$PATH_TO_INSTALL_SCRIPTS/install_via_pip.pl --file test_files/pip/version_conflict/same_pkg/step2 --ancestor-pip-package-root-path test_files/pip/version_conflict/same_pkg/build_info/packages --python-binary python3 --with-versions $DRY_RUN_OPTION")
+assert "$TEST_OUTPUT" "Dry-Run: python3 -m pip install  --no-cache-dir 'azure-common==1.1.4' 'azure-common==1.1.28'"
+echo
+
+
+echo ./install_via_pip.pl installing with ancestors but all empty
+TEST_OUTPUT=$(run_install "$PATH_TO_INSTALL_SCRIPTS/install_via_pip.pl --file test_files/pip/empty/step2 --ancestor-pip-package-root-path test_files/pip/empty/build_info/packages --python-binary python3 --with-versions $DRY_RUN_OPTION")
+assert "$TEST_OUTPUT" ""
+echo
+
+
+echo ./install_via_pip.pl installing with ancestors and correct dependency
+TEST_OUTPUT=$(run_install "$PATH_TO_INSTALL_SCRIPTS/install_via_pip.pl --file test_files/pip/no_version_conflict/dependency_already_installed/step2 --ancestor-pip-package-root-path test_files/pip/no_version_conflict/dependency_already_installed/build_info/packages --python-binary python3 --with-versions $DRY_RUN_OPTION")
+assert "$TEST_OUTPUT" "Dry-Run: python3 -m pip install  --no-cache-dir 'azure-common==1.1.4' 'azure-batch==1.0.0'"
+echo
+
+
+echo ./install_via_pip.pl installing with ancestors and dependency with wrong version must fail
```

Comment:
Hm, the tests don't directly test what we wanted to test, because they always install into a empty python environment and not the steps one after another

Reply:
We keep it that way, the conflict cases should trigger as expected, however, we should be aware that we don't test the exact cases as in the SLC. The effort to change the tests is greater than the value, if we keep in mind, that we want to replace the scripts with go.

Reply:
We should, however, add a comment that we don't test the exact cases like in the SLCs.